src/clients/SpokePoolClient.ts

```diff
@@ -201,18 +202,26 @@ export class SpokePoolClient {
   }
 
   appendMaxSpeedUpSignatureToDeposit(deposit: DepositWithBlock): DepositWithBlock {
-    const maxSpeedUp = this.speedUps[deposit.depositor]?.[deposit.depositId]?.reduce((prev, current) =>
+    const speedupsForDeposit = this.speedUps[deposit.depositor]?.[deposit.depositId];
+
+    // Find the SpeedUp with the highest relayerFeePct
+    const maxSpeedUp = speedupsForDeposit?.reduce((prev, current) =>
```

Comment:
I think this started out as a functional change, but it seems to have reverted to the pre-existing behaviour (albeit with different formatting). Should we just revert that so it disappears from the diff?

```suggestion
const maxSpeedUp = this.speedUps[deposit.depositor]?.[deposit.depositId]?.reduce((prev, current) =>
```